'Who to follow' checks to see if already followed + working follow bu…

lmurdock12 opened the pull request:
Please review code and let me know if this works for yall.

This should tackle both issues with 'Who to follow' (#59, #46)

lmurdock12 commented on Account/templates/Account/profile.html:
```diff
@@ -76,16 +72,12 @@ <h3 style="color: rgb(99, 99, 99);">@{{user.username}}</h3>
-            {% else %}
-            <button name="unfollow{{user.username}}" type="submit" style="float:right" value={{user.username}} class="follow_button">Unfollow</button>
-            {% endif %}        
+            <button name="follow" type="submit" style="float:right" value={{user.username}} class="follow_button">Follow</button>
```
'Who to follow' section should have have anyone that we already follow...so removed 'Unfollow' button

lmurdock12 commented on Account/views.py:
```diff
@@ -45,12 +45,12 @@ def profile(request, username):
             return HttpResponseRedirect(reverse('Account:login'))
 
-        if request.POST.get('follow' + profile_user.username):
+        if request.POST.get('follow'):
```
You will see that I removed the reference of  `'follow' + profile username` to just `'follow'`.

This allows a simplified approach when working with the POST requests...since we have multiple different follow buttons on a given page (that follow different accounts) we do not want to create a separate if statement to handle each post request. 

So now one POST.get('follow') will handle all the follow POST requests. We differentiate which follow button was clicked by what the actual submitted value of the follow button is. Whenever a follow button is clicked it submits the users profile name that we can reference so we know who to follow.

lmurdock12 commented on Account/views.py:
```diff
@@ -91,36 +91,31 @@ def profile(request, username):
         temp = random.choice(AllUsers)
-        while temp in rand_three or temp.username == request.user.username:
-            temp = random.choice(AllUsers)
+        #If we have an authenticated user currently logged in...check to see if we already follow this temp user
+        if request.user.is_authenticated and Follow.objects.filter(user=request.user, following=temp):
+            #Remove temp usery from Queryset if we already do
+            AllUsers = AllUsers.exclude(username=temp.username)
+            continue
+        #If temp is a correct user that we can follow..remove it from QuerySet and add it to the rand_three list
+        AllUsers = AllUsers.exclude(username=temp.username)
         rand_three.append(temp)
 
 
-    list_follow = [False, False, False]
-    if request.user.is_authenticated:
-
-
-        #Check to see if the authenticated user is already following the user
-        for i, listed_user in enumerate(rand_three, start=0):
-            temp_followed = Follow.objects.filter(following=listed_user)
-            for j in temp_followed:
-                if request.user == j.user:
-                    list_follow[i] = True
-
-    userlist = zip(rand_three, list_follow)
-
-
-    context = {'validSession':False, 'username':request.user.username, 'userdic':AllUsers, 'userlist':userlist, 
```
Simplified the approach for grabbing the three accounts to follow.

Now we grab all the users to start, and as we add a user to the rand_three dictionary we remove it from the AllUsers list (QuerySet). If a given user is found that we already follow we go ahead and remove it from AllUsers. In the end we keep looping till we either find three users that we have not already followed yet, or we ran out of users that are viable for an account to follow.

This approach also removes some complexity.

bhball22 commented:
Is there a reason we want to not show any other users if you follow everyone?

bhball22 commented:
Because right now I am not able to open the account page of any other user. So, I am unable to unfollow anyone if I wanted to

bhball22 commented:
Maybe we should comment out that information until we have a page to view those we do currently follow, when we can open their home page by clicking on tweets, and when able to be searched.

lmurdock12 commented:
> Is there a reason we want to not show any other users if you follow everyone?

I was just following the similar way the Twitter 'Who To Follow' works...where it never shows anyone that you already follow. 

> Maybe we should comment out that information until we have a page to view those we do currently follow, when we can open their home page by clicking on tweets, and when able to be searched.

Yeah that sounds good. I think for now I will just edit back in the unfollow functionality...and let 'who to follow' show accounts that the user already follows. Once we have Follow and Following tabs next sprint we can remove this. 

Making changes now.

lmurdock12 commented:
@bhball22 

I have added unfollow functionality. Let me know if this works. I also changed the section title temporarily from 'who to follow' to 'Other users on the network'. It just sounded a little weird that it was called 'who to follow' but we were seeing users we already followed.

bhball22 reviewed (approved):
I went in and fixed something simple that can be removed later
